zerver/lib/custom_zulip_transaction.py

```diff
@@ -0,0 +1,187 @@
+from collections.abc import Callable
```

Comment:
I think calling this `zerver/lib/idempotent_request.py` is probably a much better name.